src/extended.rs

```diff
@@ -1635,3 +1507,193 @@ bitflags! {
         const FP256 = 1 << 2;
     }
 }
+
+/// Processor Topology Information (LEAF=0x8000_001E).
+///
+/// # Platforms
+/// ✅ AMD ❌ Intel
+#[derive(PartialEq, Eq)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+pub struct ProcessorTopologyInfo {
+    eax: u32,
+    ebx: u32,
+    ecx: u32,
+    /// Reserved
+    _edx: u32,
+}
+
+impl ProcessorTopologyInfo {
+    pub(crate) fn new(data: CpuIdResult) -> Self {
+        Self {
+            eax: data.eax,
+            ebx: data.ebx,
+            ecx: data.ecx,
+            _edx: data.edx,
+        }
+    }
+
+    /// x2APIC ID
+    pub fn x2apic_id(&self) -> u32 {
+        self.eax
+    }
+
+    /// Core ID
+    pub fn core_id(&self) -> u8 {
```

Comment:
Maybe it might be better to stick to the AMD manual terminology here and use `compute_unit_id` instead of `core_id` 

From the manual:
"Compute unit ID. Identifies a Compute Unit, which may be one or more
physical cores that each implement one or more logical processors."

Core is a bit overloaded (never clear if hyperthreading or no?)
If we add this comment from the AMD manual to the documentation that should make things clearer too

Reply:
AMD's documentation seems to use `Compute Unit` for family 15h-16h, and `Thread` for family 17h.  
I'll add that to the comments.  

BIOS and Kernel Developer’s Guide (BKDG) for AMD Family 15h Models 00h-0Fh Processors  
Page 501: <https://www.amd.com/system/files/TechDocs/42301_15h_Mod_00h-0Fh_BKDG.pdf>

Processor Programming Reference (PPR) for AMD Family 17h Model 01h, Revision B1 Processors  
Page 82: <https://developer.amd.com/wordpress/media/2017/11/54945_PPR_Family_17h_Models_00h-0Fh.pdf>